tests/python/contrib/test_hexagon/test_software_pipeline_async.py

```diff
@@ -26,8 +26,9 @@
 
 outer = tvm.testing.parameter(8, 16)
 inner = tvm.testing.parameter(64, 128)
-scope = tvm.testing.parameter("global", "global.vtcm")
 dtype = tvm.testing.parameter("uint8", "float16")
+scope = tvm.testing.parameter("global", "global.vtcm")
+sched = tvm.testing.parameter("cache_read", "cache_read_write")
 
 
 @tvm.testing.fixture
```

Comment:
With this PR, we could technically test any n-stage pipeline, correct (doesn't have to be limited to 3-stage)?

Reply:
Correct.  This allows for any number of `cache_read` and `cache_write` stages to be lowered to Async DMA.  Note that there is a known issue when trying to do `cache_read` for an op with multiple inputs in the same stage which will be addressed in a future PR.  Future PR will modify compute on this test to be `a + b` instead of `a + 1` and add support to use Async DMA to `cache_read` both `a` and `b`.